app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -13,19 +13,16 @@ def self.call(*args)
       def call
         return if commented_on_welcome_thread? || received_notification?
 
-        welcome_broadcast
         Notification.send_welcome_notification(receiver_id, welcome_broadcast.id)
       end
 
       def received_notification?
-        welcome_broadcast
         Notification.exists?(notifiable: welcome_broadcast, user_id: receiver_id)
       end
 
       def commented_on_welcome_thread?
-        welcome_threads = Article.where("slug LIKE 'welcome-thread%'")
-        user = User.find_by(id: receiver_id)
-        Comment.where(commentable: welcome_threads, user_id: receiver_id).any? && user.created_at + 3.hours <= Time.zone.now
+        welcome_threads = Article.where("title LIKE 'Welcome Thread - %'").order(created_at: :desc).first
+        Comment.where(commentable: welcome_threads, user_id: receiver_id).any?
```

Comment:
since you already have memoized user, you could use that here and do `user: user` in the `where` clause, i think. could probably do the same on line 20, too.

Reply:
Oh, good call! I've made this change and everything still seems to work as expected. 👍